scene/3d/mesh_instance_3d.cpp

```diff
@@ -493,6 +493,93 @@ void MeshInstance3D::create_debug_tangents() {
 	}
 }
 
+Ref<ArrayMesh> MeshInstance3D::create_mesh_from_blend_shape_mix() {
+	Ref<ArrayMesh> new_baked_mesh;
+	new_baked_mesh.instantiate();
+	Ref<ArrayMesh> source_mesh = get_mesh();
+	ERR_FAIL_NULL_V_MSG(source_mesh, new_baked_mesh, "No valid source mesh with blend shapes");
+
+	for (int surface_index = 0; surface_index < source_mesh->get_surface_count(); surface_index++) {
+		const Array &source_mesh_arrays = source_mesh->surface_get_arrays(surface_index);
+
+		ERR_FAIL_COND_V(source_mesh_arrays.size() != RS::ARRAY_MAX, new_baked_mesh);
+
+		const Vector<Vector3> &source_mesh_vertex_array = source_mesh_arrays[Mesh::ARRAY_VERTEX];
+		const Vector<Vector3> &source_mesh_normal_array = source_mesh_arrays[Mesh::ARRAY_NORMAL];
+		const Vector<float> &source_mesh_tangent_array = source_mesh_arrays[Mesh::ARRAY_TANGENT];
+
+		Array new_mesh_arrays;
+		new_mesh_arrays.resize(Mesh::ARRAY_MAX);
+		for (int i = 0; i < source_mesh_arrays.size(); i++) {
+			if (i == Mesh::ARRAY_VERTEX || i == Mesh::ARRAY_NORMAL || i == Mesh::ARRAY_TANGENT) {
+				continue;
+			}
+			new_mesh_arrays[i] = source_mesh_arrays[i];
+		}
+
+		Vector<Vector3> lerped_vertex_array = source_mesh_vertex_array;
+		Vector<Vector3> lerped_normal_array = source_mesh_normal_array;
+		Vector<float> lerped_tangent_array = source_mesh_tangent_array;
+
+		const Array &blendshapes_mesh_arrays = source_mesh->surface_get_blend_shape_arrays(surface_index);
+		int blend_shape_count = source_mesh->get_blend_shape_count();
+		ERR_FAIL_COND_V(blendshapes_mesh_arrays.size() != blend_shape_count, new_baked_mesh);
+
+		for (int blendshape_index = 0; blendshape_index < blend_shape_count; blendshape_index++) {
+			float blend_weight = get_blend_shape_value(blendshape_index);
+			if (abs(blend_weight) > 0.0001) {
+				const Array &blendshape_mesh_arrays = blendshapes_mesh_arrays[blendshape_index];
+
+				const Vector<Vector3> &blendshape_vertex_array = blendshape_mesh_arrays[Mesh::ARRAY_VERTEX];
+				const Vector<Vector3> &blendshape_normal_array = blendshape_mesh_arrays[Mesh::ARRAY_NORMAL];
+				const Vector<float> &blendshape_tangent_array = blendshape_mesh_arrays[Mesh::ARRAY_TANGENT];
+
+				ERR_FAIL_COND_V(source_mesh_vertex_array.size() != blendshape_vertex_array.size(), new_baked_mesh);
+				ERR_FAIL_COND_V(source_mesh_normal_array.size() != blendshape_normal_array.size(), new_baked_mesh);
+				ERR_FAIL_COND_V(source_mesh_tangent_array.size() != blendshape_tangent_array.size(), new_baked_mesh);
+
+				for (int i = 0; i < source_mesh_vertex_array.size(); i++) {
+					const Vector3 &source_vertex = source_mesh_vertex_array[i];
+					const Vector3 &blendshape_vertex = blendshape_vertex_array[i];
+					Vector3 lerped_vertex = source_vertex.lerp(blendshape_vertex, blend_weight) - source_vertex;
```

Comment:
We might need to add another case here.
Blend shapes support two modes:
https://docs.godotengine.org/en/stable/classes/class_arraymesh.html#class-arraymesh-property-blend-shape-mode

lerp is correct for `BLEND_SHAPE_MODE_NORMALIZED` but not for `BLEND_SHAPE_MODE_RELATIVE`, which should instead be done with += I believe (we should check the shader code to make sure it's correct)